### PR TITLE
update addon developer guide

### PR DESCRIPTION
**Pull request by haoqing0110**
Signed-off-by: haoqing0110 <qhao@redhat.com>

### CODE REVIEW

File: content/en/developer-guides/addon.md [hunk 1]
```diff
@@ -55,45 +55,10 @@ You can find the example in [here](https://github.com/open-cluster-management-io
 
 ### Implement the addon manager
 
-First, you should create a folder in your Go project, for example, the folder name is `manifests`. The manifest files 
-deployed on the managed cluster are put in this folder.
+First, create your Go project, and the project should contain a `main.go` file and a folder `manifests`. `main.go` contains the Go code of the addon manager.
```
Comment (zhiweiyin318) [inline content/en/developer-guides/addon.md:58]: the name of the folder can be customized.

File: content/en/developer-guides/addon.md [hunk 2]
```diff
@@ -180,6 +201,7 @@ spec:
    displayName: Busybox Addon
    description: "busybox-addon is an example addon to deploy busybox pod on the managed cluster"
 ```
+The example folder don't have this file, is this required? what's the impact if not creat it?
```
Comment (zhiweiyin318) [inline content/en/developer-guides/addon.md:204]: the clustermanagementaddon yaml file is here https://github.com/open-cluster-management-io/addon-framework/blob/main/examples/deploy/addon/busybox/resources/busybox_clustermanagementaddon.yaml

File: content/en/developer-guides/addon.md [hunk 3]
```diff
@@ -207,6 +229,7 @@ You can also use the `clusteradm` command to enable the busybox-addon for the ma
 ```bash
 $ clusteradm addon enable --names busybox-addon --namespace open-cluster-management-agent-addon --clusters cluster1
 ```
+how can clusteradm know the addon name busybox-addon? through ClusterManagementAddOn?
```
Comment (zhiweiyin318) [inline content/en/developer-guides/addon.md:232]: my understanding is that clusteradm only create addon cr using the addon name , namesapce and cluster name.

File: content/en/developer-guides/addon.md [hunk 4]
```diff
@@ -686,6 +709,7 @@ The addon-framework supports helm charts or raw manifests as the add-on agent ma
 ### Values definition
 
 The addon-framework supports 3 add-on built-in values and 3 helm chart built-in values for helm chart add-on manifests.
+?
```
Comment (zhiweiyin318) [inline content/en/developer-guides/addon.md:712]: and Value.hubKubeConfigSecret

File: content/zh/developer-guides/addon.md [hunk 5]
```diff
@@ -55,45 +55,10 @@ You can find the example in [here](https://github.com/open-cluster-management-io
 
 ### Implement the addon manager
 
-First, you should create a folder in your Go project, for example, the folder name is `manifests`. The manifest files 
-deployed on the managed cluster are put in this folder.
+First, create your Go project, and the project should contain a `main.go` file and a folder `manifests`. `main.go` contains the Go code of the addon manager.
```
Comment (zhiweiyin318) [inline content/zh/developer-guides/addon.md:58]: the name of folder can be customized.

File: content/en/developer-guides/addon.md (hunk 1 above)
Comment (haoqing0110) [reply content/en/developer-guides/addon.md]: fixed.

File: content/en/developer-guides/addon.md (hunk 2 above)
Comment (haoqing0110) [reply content/en/developer-guides/addon.md]: fixed.

File: content/zh/developer-guides/addon.md (hunk 5 above)
Comment (haoqing0110) [reply content/zh/developer-guides/addon.md]: fixed.

**Pull request merged**